src/Sylius/Component/User/Model/CredentialsHolderInterface.php

```diff
@@ -13,62 +13,21 @@
 
 namespace Sylius\Component\User\Model;
 
-use Symfony\Component\Security\Core\User\LegacyPasswordAuthenticatedUserInterface;
-use Symfony\Component\Security\Core\User\UserInterface as SymfonyUserInterface;
+use Symfony\Component\Security\Core\User\PasswordAuthenticatedUserInterface;
 
-// Symfony 5.4
-if (\method_exists(SymfonyUserInterface::class, 'getPassword')) {
-    interface CredentialsHolderInterface
-    {
-        public function getPlainPassword(): ?string;
+interface CredentialsHolderInterface extends PasswordAuthenticatedUserInterface
+{
+    public function getPlainPassword(): ?string;
 
-        public function setPlainPassword(?string $plainPassword): void;
+    public function setPlainPassword(?string $plainPassword): void;
 
-        /**
-         * Returns the password used to authenticate the user.
-         *
-         * This should be the encoded password. On authentication, a plain-text
-         * password will be salted, encoded, and then compared to this value.
-         *
-         * @return string|null
-         */
-        public function getPassword();
+    public function setPassword(?string $encodedPassword): void;
 
-        public function setPassword(?string $encodedPassword): void;
-
-        /**
-         * Returns the salt that was originally used to encode the password.
-         *
-         * This can return null if the password was not encoded using a salt.
-         *
-         * @return string|null
-         */
-        public function getSalt();
-
-        /**
-         * Removes sensitive data from the user.
-         *
-         * This is important if, at any given point, sensitive information like
-         * the plain-text password is stored on this object.
-         */
-        public function eraseCredentials();
-    }
-    // Symfony 6
-} else {
-    interface CredentialsHolderInterface extends LegacyPasswordAuthenticatedUserInterface
-    {
-        public function getPlainPassword(): ?string;
-
-        public function setPlainPassword(?string $plainPassword): void;
-
-        public function setPassword(?string $encodedPassword): void;
-
-        /**
-         * Removes sensitive data from the user.
-         *
-         * This is important if, at any given point, sensitive information like
-         * the plain-text password is stored on this object.
-         */
-        public function eraseCredentials();
-    }
+    /**
+     * Removes sensitive data from the user.
+     *
+     * This is important if, at any given point, sensitive information like
+     * the plain-text password is stored on this object.
+     */
+    public function eraseCredentials();
```

Comment:
I didn't check how this interface is used, but do we need it at all? `Symfony\Component\Security\Core\User\UserInterface` has `eraseCredentials()` method.

Reply:
yes we have it 
```php
public function eraseCredentials(): void
{
    $this->plainPassword = null;
}
```